internal/svg/svg.go

```diff
@@ -22,19 +22,32 @@ import (
 
 // Colorize creates a new SVG from a given one by replacing all fill colors by the given color.
 func Colorize(src []byte, clr color.Color) []byte {
+	return colorizeImpl(src, clr, fyne.LogError)
+}
+
+// ColorizeError is the same as Colorize, except returning instead of logging any error.
+func ColorizeError(src []byte, clr color.Color) ([]byte, error) {
+	var err error
+	content := colorizeImpl(src, clr, func(s string, e error) {
+		err = fmt.Errorf("%s %s", s, e.Error())
+	})
+	return content, err
+}
+
+func colorizeImpl(src []byte, clr color.Color, errFn func(string, error)) []byte {
```

Comment:
This error function seems a bit strange in my opinion. Why not just use the return value of the function directly and call fyne.LogError or similar on that?

Reply:
Yeah now that I look at it again, it is a bit strange. It was to minimize both code changes and duplicated code. Fyne.LogError takes 2 arguments, so I guess the alternative is for the signature to be `func colorizeImpl(src []byte, clr color.Color) ([]byte, string, error)` so that the actual error and the context we add to it can still be passed separately to fyne.LogError. What do you think?

Reply:
Do you even need the extra string information? It just seems unintuitive from how Go error handling usually is made? We can lose a bit in the FyneLog error and add any necessary additional information in the error text?